packages/react-dom/src/__tests__/ReactServerRendering-test.js

```diff
@@ -554,4 +554,25 @@ describe('ReactDOMServer', () => {
       'The experimental Call and Return types are not currently supported by the server renderer.',
     );
   });
+
+  it('should warn when server rendering a class with a render method that does not extend React.Component', () => {
+    spyOnDevAndProd(console, 'error');
+    class ClassWithRenderNotExtended {
+      render() {
+        return <div />;
+      }
+    }
+    expect(console.error.calls.count()).toBe(0);
+    expect(() => {
+      ReactDOMServer.renderToString(<ClassWithRenderNotExtended />);
+    }).toThrow(TypeError);
+    if (__DEV__) {
+      expect(console.error.calls.count()).toBe(1);
```

Comment:
Instead of asserting like this, we use `toWarnDev()` in new code. Please see other tests.